File: TOMIMSV4/Entity/Transacciones/PreFactura/Trans_Prefactura_Det/clsBeTrans_prefactura_det.vb
Public Class clsBeTrans_prefactura_det
    Implements ICloneable

    Public Property IdTransPrefacturaDet() As Integer = 0
    Public Property IdTransPrefacturaEnc() As Integer = 0
    Public Property IdAcuerdoEnc() As Integer = 0
    Public Property Codigo_acuerdo_enc() As Integer = 0
    Public Property Codigo_producto_acuerdo_det() As String = ""
    Public Property IdAcuerdoDet() As Integer = 0
    Public Property Correlativo_detalle_acuerdo() As Integer = 0
    Public Property Numero_unidades_acuerdo_det() As Integer = 0
    Public Property Servicio As String = ""
    Public Property Descripcion As String = ""
    Public Property Monto() As Double = 0.0
    Public Property Porcentaje() As Double = 0.0
    Public Property Dias_eventos() As Integer = 0
    Public Property Valor() As Double = 0.0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Monto_Erp As Double = 0.00
    Public Property Moneda As String = ""

    Sub New()
        '#EJC: Add your constructor here... 
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Prefactura/clsBeTrans_prefactura_det.cs
namespace WMS.EntityCore.Prefactura
{
    public class clsBeTrans_prefactura_det : ICloneable
    {
        #region Propiedades

        public int IdTransPrefacturaDet { get; set; } = 0;
        public int IdTransPrefacturaEnc { get; set; } = 0;
        public int IdAcuerdoEnc { get; set; } = 0;
        public int Codigo_acuerdo_enc { get; set; } = 0;
        public string Codigo_producto_acuerdo_det { get; set; } = "";
        public int IdAcuerdoDet { get; set; } = 0;
        public int Correlativo_detalle_acuerdo { get; set; } = 0;
        public decimal Numero_unidades_acuerdo_det { get; set; } = 0;
        public double Servicio { get; set; } = 0;
        public string Descripcion { get; set; } = "";
        public double Monto { get; set; } = 0;
        public double Porcentaje { get; set; } = 0;
        public int Dias_eventos { get; set; } = 0;
        public double Valor { get; set; } = 0;
        public string User_agr { get; set; } = "";
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public string User_mod { get; set; } = "";
        public DateTime Fec_mod { get; set; } = DateTime.Now;
        public bool Activo { get; set; } = true;
        public double Monto_Erp { get; set; } = 0;

        #endregion

        #region Constructores

        public clsBeTrans_prefactura_det()
        {
        }

        public clsBeTrans_prefactura_det(
            int idTransPrefacturaDet,
            int idTransPrefacturaEnc,
            int idAcuerdoEnc,
            string codigoProducto)
        {
            IdTransPrefacturaDet = idTransPrefacturaDet;
            IdTransPrefacturaEnc = idTransPrefacturaEnc;
            IdAcuerdoEnc = idAcuerdoEnc;
            Codigo_producto_acuerdo_det = codigoProducto ?? "";
        }

        #endregion

        #region ICloneable

        public object Clone()
        {
            return this.MemberwiseClone();
        }

        #endregion
    }
}